### PR TITLE
Add support for multi-instance in properties file.

### PR DESCRIPTION
**Pull request by andy-scollard**
Example multi-instance usage:

git.host=<host1>, <host2>
git.username=<user1>, <user2>
git.password=<password1>, <password2>

Passwords are Base64 encrypted

### CODE REVIEW

File: src/main/java/com/capitalone/dashboard/collector/PullRequestCollector.java [hunk 1]
```diff
@@ -60,14 +59,22 @@
      * @param status            open/merged
      *
      * @return
-     * @throws EncryptionException
+     * @throws Exception
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/PullRequestCollector.java]: Technically you are suppressing the Exception by just logging the message. Can you remove this line?

File: src/main/java/com/capitalone/dashboard/collector/SCMHttpRestClient.java [hunk 2]
```diff
@@ -21,27 +21,31 @@
...
+        /*if (StringUtils.isEmpty(userId) || StringUtils.isEmpty(password)) {
+            id = settings.getUsername();
+            try {
+                secret = new String(Base64.decodeBase64(settings.getPassword()));
+            } catch (Exception e) {
+                LOG.error(e.getMessage());
+            }
+        }*/
+        // Basic Auth only.
+        if (!"".equals(id) && !"".equals(secret)) {
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/SCMHttpRestClient.java]: Can we use `StringUtils.isNotEmpty()` here?

File: src/test/java/com/capitalone/dashboard/collector/DefaultBitbucketServerClientTest.java [hunk 3]
```diff
@@ -129,17 +124,19 @@ public void testGetCommits() throws IOException, URISyntaxException {
     }
 
     @Test
-    public void testCommits() throws EncryptionException, URISyntaxException {
+    public void testCommits() throws Exception, URISyntaxException {
         // Note that there always is paging even if results only take 1 page
         String jsonResponse1 = "{ \"values\": [] }";
-        settings.setKey("abcdefghijklmnopqrstuvwxyz1234567");
-        String encPassword = Encryption.encryptString("password", settings.getKey());
+       // settings.setKey("abcdefghijklmnopqrstuvwxyz1234567");
```
Comment (rvema) [inline src/test/java/com/capitalone/dashboard/collector/DefaultBitbucketServerClientTest.java]: please remove the commented lines of code.

File: src/test/java/com/capitalone/dashboard/collector/PullRequestCollectorTest.java [hunk 4]
```diff
@@ -54,14 +54,15 @@
 
     @Test
     public void testGetPullRequests() throws IOException, URISyntaxException, EncryptionException {
-        given(settings.getKey()).willReturn("abcdefghijklmnopqrstuvwxyz1234567");
-        String encPassword = Encryption.encryptString("password", settings.getKey());
-        given(settings.getPassword()).willReturn(encPassword);
+        //given(settings.getKey()).willReturn("abcdefghijklmnopqrstuvwxyz1234567");
```
Comment (rvema) [inline src/test/java/com/capitalone/dashboard/collector/PullRequestCollectorTest.java]: please remove the commented lines of code

File: src/test/java/com/capitalone/dashboard/collector/PullRequestCollectorTest.java [hunk 5]
```diff
@@ -89,30 +90,31 @@ public void testGetPullRequests() throws IOException, URISyntaxException, Encryp
...
 
     }
 
     @Test
-    public void testGetPullRequests_RecentlyUpdated() throws IOException, URISyntaxException, EncryptionException {
-        given(settings.getKey()).willReturn("abcdefghijklmnopqrstuvwxyz1234567");
-        String encPassword = Encryption.encryptString("password", settings.getKey());
-        given(settings.getPassword()).willReturn(encPassword);
+    public void testGetPullRequests_RecentlyUpdated() throws IOException, URISyntaxException, Exception {
+//        given(settings.getKey()).willReturn("abcdefghijklmnopqrstuvwxyz1234567");
```
Comment (rvema) [inline src/test/java/com/capitalone/dashboard/collector/PullRequestCollectorTest.java]: please remove all commented lines of code

File: src/test/java/com/capitalone/dashboard/collector/SCMHttpRestClientTest.java [hunk 6]
```diff
@@ -88,23 +88,24 @@ public void makeRestCallAndtestBasicAuthenticationForGlobalCredentials() throws
                 classArgumentCaptor.capture()))
                 .willReturn(responseEntity);
 
-        given(settings.getKey()).willReturn("WI9XFb8cW9bxeS/fwdka0LBXFbqtwvEa");
-        given(settings.getPassword()).willReturn("password");
-        given(settings.getUsername()).willReturn("user");
+//        given(settings.getKey()).willReturn("WI9XFb8cW9bxeS/fwdka0LBXFbqtwvEa");
```
Comment (rvema) [inline src/test/java/com/capitalone/dashboard/collector/SCMHttpRestClientTest.java]: please remove the commented lines of code

**Review by rvema — CHANGES_REQUESTED** [review]
please review the comments and address them! Thanks for your contribution

File: src/main/java/com/capitalone/dashboard/collector/PullRequestCollector.java (hunk 1 above)
Comment (andy-scollard) [reply src/main/java/com/capitalone/dashboard/collector/PullRequestCollector.java]: done

File: src/main/java/com/capitalone/dashboard/collector/SCMHttpRestClient.java (hunk 2 above)
Comment (andy-scollard) [reply src/main/java/com/capitalone/dashboard/collector/SCMHttpRestClient.java]: done

File: src/test/java/com/capitalone/dashboard/collector/PullRequestCollectorTest.java (hunk 4 above)
Comment (andy-scollard) [reply src/test/java/com/capitalone/dashboard/collector/PullRequestCollectorTest.java]: done

File: src/test/java/com/capitalone/dashboard/collector/PullRequestCollectorTest.java (hunk 5 above)
Comment (andy-scollard) [reply src/test/java/com/capitalone/dashboard/collector/PullRequestCollectorTest.java]: done

File: src/test/java/com/capitalone/dashboard/collector/SCMHttpRestClientTest.java (hunk 6 above)
Comment (andy-scollard) [reply src/test/java/com/capitalone/dashboard/collector/SCMHttpRestClientTest.java]: done

**Review by rvema — APPROVED** [review]
lgtm

**Pull request merged**